App/iOS/Delegates/SceneDelegate.swift

```diff
@@ -369,53 +355,126 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
-      animations: {
-        window.overrideUserInterfaceStyle = self.expectedThemeOverride
-      }, completion: nil)
+  private func updateTheme(for scene: UIWindowScene) {
+    scene.windows.forEach { window in
+      UIView.transition(
+        with: window, duration: 0.15, options: [.transitionCrossDissolve],
+        animations: {
+          window.overrideUserInterfaceStyle = self.expectedThemeOverride(for: scene)
+        }, completion: nil)
+    }
   }
 }
 
 extension SceneDelegate {
-  private func createBrowserWindow(scene: UIWindowScene, braveCore: BraveCoreMain, profile: Profile, diskImageStore: DiskImageStore?, migration: Migration?) -> BrowserViewController {
-    // Make sure current private browsing flag respects the private browsing only user preference
-    PrivateBrowsingManager.shared.isPrivateBrowsing = Preferences.Privacy.privateBrowsingOnly.value
+  private func createBrowserWindow(scene: UIWindowScene, braveCore: BraveCoreMain, profile: Profile, diskImageStore: DiskImageStore?, migration: Migration?, rewards: Brave.BraveRewards, userActivity: NSUserActivity?) -> BrowserViewController {
+
+    let privateBrowsingManager = PrivateBrowsingManager()
 
     // Don't track crashes if we're building the development environment due to the fact that terminating/stopping
     // the simulator via Xcode will count as a "crash" and lead to restore popups in the subsequent launch
     let crashedLastSession = !Preferences.AppState.backgroundedCleanly.value && AppConstants.buildChannel != .debug
+    
+    // Store the scene's activities
+    let windowId: UUID
+    let isPrivate: Bool
+    var userActivity = userActivity
+    
+    if let userActivity = userActivity {
+      // Restore the scene with the WindowID from the User-Activity
+      
+      let windowIdString = userActivity.userInfo?["WindowID"] as? String ?? ""
+      windowId = UUID(uuidString: windowIdString) ?? UUID()
+      isPrivate = userActivity.userInfo?["isPrivate"] as? Bool == true
+      privateBrowsingManager.isPrivateBrowsing = isPrivate
+      
+      // Create a new session window
+      SessionWindow.createWindow(isPrivate: isPrivate, isSelected: true, uuid: windowId)
+      
+      scene.userActivity = userActivity
+      scene.session.userInfo?["WindowID"] = windowId
+      scene.session.userInfo?["isPrivate"] = isPrivate
+    } else if let sceneWindowId = scene.session.userInfo?["WindowID"] as? String,
+              let sceneIsPrivate = scene.session.userInfo?["isPrivate"] as? Bool,
+              let windowUUID = UUID(uuidString: sceneWindowId) {
+      
+      // Restore the scene from the Session's User-Info WindowID
+      
+      windowId = windowUUID
+      isPrivate = sceneIsPrivate
+      privateBrowsingManager.isPrivateBrowsing = sceneIsPrivate
+      
+      scene.userActivity = BrowserState.userActivity(for: windowId, isPrivate: isPrivate)
+    } else {
+      // Should NOT be possible to get here.
```

Comment:
I want to keep this code here because it does handle it gracefully. I simulated it by allowing the app-icon to open multiple windows. When that app-icon thing is disabled, then it'll theoretically never hit this condition. But I want to keep this code here in case we ever had menus or anything to the icon, so it'll handle such cases when the app is open or closed :)